src/broker/content-cache.c

```diff
@@ -858,12 +860,33 @@ static struct flux_msg_handler_spec handlers[] = {
     FLUX_MSGHANDLER_TABLE_END,
 };
 
+static flux_msg_handler_t *add_handler (flux_t *h, char *topic, uint32_t roles,
+                                        flux_msg_handler_f cb, void *arg)
+{
+    flux_msg_handler_t *w;
+    struct flux_match match = FLUX_MATCH_REQUEST;
+
+    match.topic_glob = topic;
+    if (!(w = flux_msg_handler_create (h, match, cb, arg)))
+        return NULL;
+    flux_msg_handler_allow_rolemask (w, roles);
+    flux_msg_handler_start (w);
+    return w;
+}
+
+
 int content_cache_set_flux (content_cache_t *cache, flux_t *h)
 {
     cache->h = h;
 
     if (flux_msg_handler_addvec (h, handlers, cache) < 0)
-	return -1;
+        return -1;
+    if (!(cache->load_w = add_handler (h, "content.load", FLUX_ROLE_USER,
+                                       content_load_request, cache)))
+        return -1;
+    if (!(cache->store_w = add_handler (h, "content.store", FLUX_ROLE_USER,
+                                        content_store_request, cache)))
+        return -1;
     if (flux_get_rank (h, &cache->rank) < 0)
```

Comment:
Question: I was looking at this commit as an example of converting a service to other users than owner, but this is a bit confusing since `content.load` and `content.store` are still in the `handlers` vector. Is that on purpose because the subsequent handlers will override the previously registered versions, or just an oversight?

Can `flux_msg_handler_allow_rolemask` be called *after* `flux_msg_handler_start`? If so, a call to return the watcher from a handler vector might be really handy, eg..

```C
flux_msg_handler_t *w = flux_msg_handler_vec_find (handlers, "content.load");
if (w)
    flux_msg_handler_allow_rolemask (w, FLUX_ROLE_USER);
```

If not,  a version of `addvec` that directly takes allowed roles would be nice.